kie-wb-smoke-tests/src/test/java/org/kie/smoke/wb/util/RestUtil.java

```diff
@@ -37,270 +48,359 @@
-     * @param responseType The {@link Class} of the return type
-     * @param status The HTTP status that the request should return
-     * @return The actual entity returned by the request: this can be null
-     */
-    public static <T> T  delete(ClientRequest restRequest, MediaType mediaType, int status, Class<T> responseType) {
-        return httpMethodReturnType(DEL, restRequest, mediaType, status, responseType);
+    private static void logOp( String op, String uri ) {
+        logger.debug(op + " " + uri);
     }
 
-    /**
-     * Call the GET HTTP method on the given request and return the actual entity returned
-     * @param restRequest The {@link ClientRequest} instance with the request information
-     * @param mediaType The {@link MediaType} that should be used with the request
-     * @param responseType The {@link Class} of the return type
-     * @param status The HTTP status that the request should return
-     * @return The actual entity returned by the request: this can be null
-     */
-    public static <T> T get(ClientRequest restRequest, MediaType mediaType, int status, Class<T> responseType) {
-        return httpMethodReturnType(GET, restRequest, mediaType, status, responseType);
+    private static void logOp( String op, Object entity, String uri ) {
+        logger.debug(op + " [" + entity.getClass().getSimpleName() + "] " + uri);
     }
 
-    /**
-     * Call the POST HTTP method on the given request and return the actual entity returned
-     * @param restRequest The {@link ClientRequest} instance with the request information
-     * @param mediaType The {@link MediaType} that should be used with the request
-     * @param responseType The {@link Class} of the return type
-     * @param status The HTTP status that the request should return
-     * @return The actual entity returned by the request: this can be null
-     */
-    public static <T> T post(ClientRequest restRequest, MediaType mediaType, int status, Class<T> responseType) {
-        return httpMethodReturnType(POST, restRequest, mediaType, status, responseType);
-    }
+    // Public Helper methods ------------------------------------------------------------------------------------------------------
 
-    /**
-     * Call the POST HTTP method on the given request and then releases the connection.
-     * @param restRequest The {@link ClientRequest} instance with the request information
-     * @param mediaType The {@link MediaType} that should be used with the request
-     * @param status The HTTP status that the request should return
-     * @return The actual entity returned by the request: this can be null
-     */
-    public static void post(ClientRequest restRequest, MediaType mediaType, int status) {
-        ClientResponse<?> responseObj = httpMethodReturnResponse(POST, restRequest, mediaType, status);
-        responseObj.releaseConnection();
+    public static void failAndLog( String msg, Exception e ) {
```

Comment:
Nit-picking here...but based on what the method does, I think it should be called `logAndFail()`